dotCMS/src/main/java/com/dotcms/content/elasticsearch/business/ESContentFactoryImpl.java

```diff
@@ -679,64 +681,87 @@ private void deleteTreesForInodes(List<String> inodes) throws DotDataException {
-        Date date = calendar.getTime();
-        //Because of the way Oracle databases handle dates,
-        //this string is converted to Uppercase.This does
-        //not cause a problem with the other databases
-        DotConnect dc = new DotConnect();
+        final Date date = calendar.getTime();
 
-        String countSQL = ("select count(*) as count from contentlet");
+        DotConnect dc = new DotConnect();
+        final String countSQL = "select count(*) as count from contentlet";
         dc.setSQL(countSQL);
         List<Map<String, String>> result = dc.loadResults();
-        int before = Integer.parseInt(result.get(0).get("count"));
-
-        String deleteContentletSQL = "delete from contentlet where identifier<>'SYSTEM_HOST' and mod_date < ? " +
-        "and not exists (select * from contentlet_version_info where working_inode=contentlet.inode or live_inode=contentlet.inode)";
-        dc.setSQL(deleteContentletSQL);
-        dc.addParam(date);
-        dc.loadResult();
-
-        String deleteOrphanInodes="delete from inode where type='contentlet' and idate < ? and inode not in (select inode from contentlet)";
-        dc.setSQL(deleteOrphanInodes);
+        final int before = Integer.parseInt(result.get(0).get("count"));
+        dc = new DotConnect();
+        dc.setSQL("SELECT inode FROM contentlet WHERE identifier <> 'SYSTEM_HOST' AND mod_date < ? AND NOT EXISTS " +
+                "(SELECT working_inode, live_inode FROM contentlet_version_info WHERE working_inode = contentlet.inode OR " +
+                "live_inode = contentlet.inode)");
         dc.addParam(date);
-        dc.loadResult();
-
+        result = dc.loadResults();
+        int oldInodesCount = result.size();
+        if (oldInodesCount > 0) {
+            final List<String> inodeList = new ArrayList<>();
+            for (final Map<String, String> row : result) {
+                inodeList.add(row.get("inode").toString());
+            }
+            final int splitAt = 100;
+            // Split all records into lists of size 'truncateAt'
+            final List<List<String>> inodesToDelete = Lists.partition(inodeList, splitAt);
+            final List<String> queries = Lists.newArrayList("DELETE FROM contentlet WHERE inode IN (?)",
+                    "DELETE FROM inode WHERE inode IN (?)");
+            for (final String query : queries) {
+                for (final List<String> inodes : inodesToDelete) {
+                    dc = new DotConnect();
+                    // Generate the "(?,?,?...)" string depending of the number of inodes
+                    final String parameterPlaceholders = DotConnect.createParametersPlaceholder(inodes.size());
+                    dc.setSQL(query.replace("?", parameterPlaceholders));
+                    for (final String inode : inodes) {
+                        dc.addParam(inode);
+                    }
+                    dc.loadResult();
+                }
+            }
+        }
```

Comment:
I think all this logic should go into a private method

https://github.com/dotCMS/core/pull/18507/files#diff-0d54bf61089532e67d512211809b7a69R710-R727

Reply:
Fixed.